### PR TITLE
RetroAchievements Bugfix: Fixed Missing DoFrame Call

### PR DESCRIPTION
**Pull request by LillyJadeKatrin**
Somewhere in the process of getting the memory peeking right for achievements, the AchievementManager call to DoFrame went missing. This restores it properly.

### CODE REVIEW

**Comment by AdmiralCurtiss** [comment]
I pointed out flaws in the function itself (related to the rich presence) that should be fixed before the call is added.

File: Source/Core/Core/AchievementManager.h [hunk 1]
```diff
@@ -109,7 +109,7 @@ class AchievementManager
   u32 m_game_id = 0;
   rc_api_fetch_game_data_response_t m_game_data{};
   bool m_is_game_loaded = false;
-  u64 m_last_ping_time = 0;
+  time_t m_last_ping_time = std::time(nullptr);
```
Comment (AdmiralCurtiss) [inline Source/Core/Core/AchievementManager.h:112]: Needs an `#include <ctime>`

Comment (AdmiralCurtiss) [reply Source/Core/Core/AchievementManager.h:112]: Also I don't think you want to initialize this on class construction, initialize it on game start instead.

Comment (LillyJadeKatrin) [reply Source/Core/Core/AchievementManager.h:112]: Alternate idea, what if I initialize it to zero and reset it to zero at game close? That way when the first frame hits, it's already been more than two minutes and immediately generates the first RP.

**Pull request merged**